### PR TITLE
MainSolver: Keep only one way to assert formula

### PR DESCRIPTION
**Pull request by blishko**

### CODE REVIEW

**Review by BritikovKI — APPROVED** [review]
Lgtmt

File: src/api/Interpret.cc [hunk 1]
```diff
@@ -217,29 +217,20 @@ void Interpret::interp(ASTNode& n) {
...
-                        if (status == s_Error)
-                            notify_formatted(true, "Error");
-                        else if (status == s_Undef)
-                            notify_success();
-                        else if (status == s_False)
+                        if (logic->getSortRef(tr) != logic->getSort_bool()) {
+                            auto errorMessage = "Top-level assertion sort must be Bool, got " + logic->printSort(logic->getSortRef(tr));
+                            notify_formatted(true, errorMessage.c_str());
+                        } else {
+                            main_solver->insertFormula(tr);
```
Comment (aehyvari) [inline src/api/Interpret.cc:231]: Maybe
```suggestion
                        try {
                            main_solver->insertFormula(tr);
                            notify_success();
                        } catch (OsmtApiException const & e) {
                            notify_formatted(true, e.msg.c_str());
                        }

```
(modulo my c++ syntax & other errors...)

Comment (blishko) [reply src/api/Interpret.cc]: Fixed and added a test for this as well, just in case.

**Review by aehyvari — APPROVED** [review]
LGTM!

**Pull request merged**